aspnet/mvc/overview.rst

```diff
@@ -1,8 +1,313 @@
+
+
+.. code-block:: c#
+
+  routes.MapRoute(name: "Default", template: "{controller=Home}/{action=Index}/{id?}");
+
+*Attribute routing* enables you to specify routing information by decorating your controllers and actions with attributes that define your application's routes. This means that your route definitions are placed next to the controller and action with which they're associated.
+
+.. code-block:: c#
+  :emphasize-lines: 1,4
+
+  [Route("api/[controller]")]
+  public class ProductsController : Controller
+  {
+    [HttpGet("{id}")]
+    public IActionResult GetProduct(int id)
+    {
+      ...
+    }
+  }
+
+Model binding
+^^^^^^^^^^^^^
+
+ASP.NET Core MVC :doc:`model binding </mvc/models/model-binding>` converts client request data  (form values, route data, query string parameters, HTTP headers) into objects that the controller can handle. As a result, your controller logic doesn't have to do the work of figuring out the incoming request data; it simply has the data as parameters to its action methods.
+
+.. code-block:: C#
+
+  public async Task<IActionResult> Login(LoginViewModel model, string returnUrl = null) { ... }
+
+Model validation
+^^^^^^^^^^^^^^^^
+
+In addition to binding request data to model objects, ASP.NET Core MVC provides a powerful :doc:`validation <mvc/models/validation>` system. Simply add validation attributes to your model types using data annotation attributes and then check for model errors in your controller action.
+
+.. code-block:: c#
+  :emphasize-lines: 4-5,8-9
+
+  using System.ComponentModel.DataAnnotations;
+  public class LoginViewModel
+  {
+      [Required]
+      [EmailAddress]
+      public string Email { get; set; }
+
+      [Required]
+      [DataType(DataType.Password)]
+      public string Password { get; set; }
+
+      [Display(Name = "Remember me?")]
+      public bool RememberMe { get; set; }
+  }
+
+A controller action:
+
+.. code-block:: c#
+  :emphasize-lines: 3
+  
+  public async Task<IActionResult> Login(LoginViewModel model, string returnUrl = null)
+  {
+      if (ModelState.IsValid)
+      {
+        // work with the model
+      }
+      // If we got this far, something failed, redisplay form
+      return View(model);
+  }
+
+The framework will handle validating request data both on the client and on the server. Validation logic specified on model types is added to the rendered views as unobtrusive annotations and is enforced in the browser via `jQuery Validation <http://jqueryvalidation.org/>`__.
+
+Dependency injection
+^^^^^^^^^^^^^^^^^^^^
+
+ASP.NET Core MVC leverages ASP.NET Core's built-in support for :doc:`dependency injection </fundamentals/dependency-injection>`. :doc:`Controllers </mvc/controllers/dependency-injection>` can request needed services through their constructors, allowing them to follow the `Explicit Dependencies Principle <http://deviq.com/explicit-dependencies-principle/>`_:
```

Comment:
I think we should try not to use ASP.NET Core followed by ASP.NET Core MVC - it becomes tedious and repetitive.  HOw about:

ASP.NET Core has built-in support for :doc:`DI